### PR TITLE
feat: update ipfsd-ctl to 4.x

### PR DESCRIPTION
**Pull request by hacdias**
Superseeds #1302. Closes #1412.

I'm starting this on a new branch with the version 3.x. I'm using a lot o @lidel's work on that PR.

### CODE REVIEW

**Comment by hacdias** [comment]
Seems to fail only on AppVeyor. It does seem to be something to do with the output parsing.

**Comment by hacdias** [comment]
@lidel it seems the tests failed on Windows because of some beautiful characters known as CR. `.trim()` seems to have solved the issue.

**Comment by lidel** [comment]
_I'll take a look at this as soon as I figure out how to get tray icon back (#1153) :upside_down_face:_

**Comment by hacdias** [comment]
Just upgraded some files where I saw the API changed. Completely missed those!

**Comment by hacdias** [comment]
Update to 4.x blocked: https://github.com/ipfs/js-ipfsd-ctl/issues/501

**Comment by lidel** [comment]
@hacdias should be fixed in js-ipfsd-ctl [v4.0.1](https://github.com/ipfs/js-ipfsd-ctl/releases/tag/v4.0.1) – mind taking a look?

**Comment by hacdias** [comment]
@lidel seems to be working!

**Comment by hacdias** [comment]
Just rebased on top of master

File: src/daemon/index.js [hunk 1]
```diff
@@ -78,21 +78,19 @@ module.exports = async function (ctx) {
     const log = logger.start('[ipfsd] stop daemon', { withAnalytics: 'DAEMON_STOP' })
     updateStatus(STATUS.STOPPING_STARTED)
 
-    if (!fs.pathExists(join(ipfsd.repoPath, 'config'))) {
+    if (!fs.pathExistsSync(join(ipfsd.path, 'config'))) {
```
[the remark this answers is not in the document]
Comment (hacdias) [reply src/daemon/index.js:81]: Oopsie. I knew that but completely missed. Probably forgot adding an `await`. Thanks!

File: src/daemon/config.js [hunk 2]
```diff
@@ -184,8 +192,13 @@ async function checkPorts (ipfsd) {
   const apiPort = parseInt(configApiMa.nodeAddress().port, 10)
   const gatewayPort = parseInt(configGatewayMa.nodeAddress().port, 10)
 
-  const freeGatewayPort = await portfinder.getPortPromise({ port: gatewayPort, stopPort: gatewayPort + 100 })
-  const freeApiPort = await portfinder.getPortPromise({ port: apiPort, stopPort: apiPort + 100 })
+  const findFreePort = async (port, from) => {
+    port = Math.max(port, from, 1024)
```
Comment (guanzo) [inline src/daemon/config.js:196]: Hello, this `Math.max` call prevents me from setting a gateway port lower than 8080, and an api port lower than 5001. When I edit the config file and restart ipfs-desktop, it says the port is busy and suggests a new one. 

For example if I try to set gateway port to 6437, then

```js
const gatewayPort = 6437 // read from config file
const freeGatewayPort = await findFreePort(gatewayPort , 8080) // returns 8080

// evaluates to true, even though 6437 is free.
const busyGatewayPort = gatewayPort !== freeGatewayPort 

// I get a popup saying "The port 6437 is not available. Do you want to use 8080 instead?"
```

**Pull request merged**